src/leet2git/leet2git.py

```diff
@@ -0,0 +1,293 @@
+@click.pass_obj
+def remove_question(cm: ConfigManager, id: int):
+    """Delete a question and its files
+
+    Args:
+        id (int): the question id
+    """
+    qdb: QuestionDB = QuestionDB(cm.config)
+    qdb.load()
+    if qdb.check_if_exists(id):
+        data = qdb.get_data()[id]
+        try:
+            os.remove(data.file_path)
+            os.remove(data.test_file_path)
+        except FileNotFoundError as e:
+            click.secho(e.args)
+        qdb.delete_question(id)
+        qdb.save()
+        # update readme
+        rh = ReadmeHandler(cm.config)
+        rh.build_readme(qdb.get_sorted_list(sort_by="creation_time"))
+        click.secho(f"The question {id} was removed.")
+    else:
+        click.secho(f"The question {id} could not be found!")
+
+
+@leet2git.command()
+@click.option(
+    "--source-repository", "-s", default="", help="the path to the folder where the code will be saved"
+)
+@click.option("--language", "-l", default="python3", help="the default language")
+@click.option("--create-repo", "-c", is_flag=True, help="generates a git repository")
+@click.pass_obj
+def init(cm: ConfigManager, source_repository: str, language: str, create_repo: bool):
+    """Creates a new configuration file and can generate a git repository.
+    \f
+    Args:
+        source_repository (str, optional): the path to the folder where the code will be saved. Defaults to "".
+        language (str, optional): the default language. Defaults to "python3".
+        create_repo (bool, optional): generates a git repository. Defaults to False.
+    """
+    reset_config(cm, source_repository, language, load_old=False)
+    cm.load_config()
+    if create_repo:
+        data = QuestionData(language=cm.config["language"])
+        file_handler = FileHandler(data, cm.config)
+        file_handler.generate_repo(cm.config["source_path"])
+
+
+@leet2git.command()
+@click.option(
+    "--source-repository", "-s", default="", help="the path to the folder where the code will be saved"
+)
+@click.option("--language", "-l", default="python3", help="the default language")
+@click.option(
+    "--soft/--hard",
+    default=True,
+    help="A soft reset only erases the database. A hard reset also erase the files.",
+)
+@click.pass_obj
+def reset(cm: ConfigManager, source_repository: str, language: str, soft: bool):
+    """Reset the configuration file
+    \f
+    Args:
+        source_repository (str, optional): the path to the folder where the code will be saved. Defaults to "".
+        language (str, optional): the default language. Defaults to "python3".
+        soft(bool, optional): the reset hardness. Defaults to soft.
+    """
+    if not soft:
+        try:
+            click.confirm(
+                f"This will delete EVERY FILE inside the {cm.config['source_path']} folder. Still want to proceed?",
+                abort=True,
+            )
+        except Abort:
+            return
+
+        shutil.rmtree(cm.config["source_path"])
```

Comment:
while leet2git reset --soft works fine, I think we should reconsider about reset --hard
First of all, in windows, access error occurs with reset --hard.
Also, I think reset should be function of removing question DB, source codes and test codes rather than removing everything inside the repo folder.